### PR TITLE
Block library: Refactor Paragraph block to use class name for text align

### PR DESCRIPTION
**Pull request by gziolo**
## Description
Related: https://github.com/WordPress/gutenberg/issues/16027, https://github.com/WordPress/gutenberg/issues/15751, https://github.com/WordPress/gutenberg/pull/16777, and https://github.com/WordPress/gutenberg/pull/16779.

This PR follows-up #16035.

Before:
```html
<!-- wp:core/paragraph {"align":"right"} -->
<p style="text-align:right;">... like this one, which is separate from the above and right aligned.</p>
<!-- /wp:core/paragraph -->
```

After:
```html
<!-- wp:paragraph {"align":"right"} -->
<p class="has-text-align-right">... like this one, which is separate from the above and right aligned.</p>
<!-- /wp:paragraph -->
```

## How has this been tested?

Using one of the existing branches (`master` probably) add a few Paragraph blocks and set different text alignments and save your post. Open the same post with this branch and ensure that it still works

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate.

### CODE REVIEW

**Comment by gziolo** [comment]
When I go to [Gutenberg demo page](http://localhost:8888/wp-admin/post-new.php?gutenberg-demo) locally I see lots of warnings on JS console. It looks like validation runs twice, but more importantly in this case I need to update HTML source of the demo content. I plan to open a related PR where I will improve the feedback shared on the JS console. It clearly is wrong as is, we should only let developers know that post was properly converted and show what was before and how it is coded from now on.

Screen Shot 2019-07-29 at 17 00 52

@youknowriad, @aduth, @mtias or @mcsf, do you know why validation is triggered twice?

**Comment by aduth** [comment]
@gziolo If I had to guess without looking, we have a few issues present in development environments due to how React fires `constructor` twice in strict mode:

https://reactjs.org/docs/strict-mode.html#detecting-unexpected-side-effects

For us, likely most problematic here:

https://github.com/WordPress/gutenberg/blob/f598210ecf3944fd9a66e5266399853b729c7293/packages/editor/src/components/provider/index.js#L57

**Comment by youknowriad** [comment]
Validation also runs for every deprecated version until it finds the "correct" version to use. I think the path forward would be to not show any error unless there's no corresponding deprecated version.

**Comment by gziolo** [comment]
> @gziolo If I had to guess without looking, we have a few issues present in development environments due to how React fires `constructor` twice in strict mode:
> 
> > Strict mode can’t automatically detect side effects for you, but it can help you spot them by making them a little more deterministic. This is done by intentionally double-invoking the following methods:
> > 
> > * Class component constructor method
> > * [...]
> 
> https://reactjs.org/docs/strict-mode.html#detecting-unexpected-side-effects
> 
> For us, likely most problematic here:
> 
> https://github.com/WordPress/gutenberg/blob/f598210ecf3944fd9a66e5266399853b729c7293/packages/editor/src/components/provider/index.js#L57

@aduth, you are totally correct. I tried with `SCRIPT_DEBUG` both enabled and disabled. I can see it triggered twice only in development mode which confirms what you shared. I guess we can ignore it then or refactor this provider to use hooks.

> Validation also runs for every deprecated version until it finds the "correct" version to use. I think the path forward would be to not show any error unless there's no corresponding deprecated version.

@youknowriad – I'm working on it in separate PR 👍

**Comment by gziolo** [comment]
Applying `Needs Dev Note` as this might have an impact on themes which used the presence of style as a way to change the visual aspect of the block.

File: packages/block-library/src/paragraph/deprecated.js [hunk 1]
```diff
@@ -11,7 +11,7 @@ import {
 	RawHTML,
 } from '@wordpress/element';
 import {
-	getColorClassName,
+	getColorClassName, getFontSizeClass,
```
Comment (youknowriad) [inline packages/block-library/src/paragraph/deprecated.js]: Maybe in a new line?

**Review by youknowriad — APPROVED** [review]

File: packages/block-library/src/paragraph/deprecated.js (hunk 1 above)
Comment (gziolo) [reply packages/block-library/src/paragraph/deprecated.js]: Sure, will fix 👍 

PhpStorm adds those imports now, I wasn't aware that it adds in the same line 😅

**Comment by gziolo** [comment]
> Any other block left with the inline styles for these?

It is the last one I think. Well, at least I don't have more PRs opened :)

We decided we don't add text-align to List block but it would be a new feature.

File: packages/block-library/src/paragraph/deprecated.js (hunk 1 above)
Comment (gziolo) [reply packages/block-library/src/paragraph/deprecated.js]: 1bab5b3a2ab1b738449cf6f12dd55c7bfc4dc939 ✅

**Pull request merged**